src/gui/DatabaseOpenWidget.cpp

```diff
@@ -227,23 +220,23 @@ void DatabaseOpenWidget::openDatabase()
     }
 
     if (ok) {
-#ifdef WITH_XC_TOUCHID
-        QHash<QString, QVariant> useTouchID = config()->get(Config::UseTouchID).toHash();
-
-        // check if TouchID can & should be used to unlock the database next time
-        if (m_ui->checkTouchID->isChecked() && TouchID::getInstance().isAvailable()) {
-            // encrypt and store key blob
-            if (TouchID::getInstance().storeKey(m_filename, PasswordKey(m_ui->editPassword->text()).rawKey())) {
-                useTouchID.insert(m_filename, true);
-            }
-        } else {
-            // when TouchID not available or unchecked, reset for the current database
-            TouchID::getInstance().reset(m_filename);
-            useTouchID.insert(m_filename, false);
+        if (m_ui->quickUnlockCheckbox->isChecked() && !m_ui->editPassword->text().isEmpty()) {
```

Comment:
So how about the middle credential box get wrapped in a QStackedWidget and when Quick Unlock is available for use it shows the second stack which is the UNLOCK / CLEAR buttons. I really like the idea.

Reply:
What do we need the key file for if we already have the transformed key stored in memory? Only hiding the password field is probably even more confusing.

Reply:
We don't store the transformed key, we store the password hash. If you stored the transformed key you would be unable to unlock the database if it is saved since the key will change.